Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1559,7 +1559,20 @@ bool buildStateReferences
                         )
                     );
 
-                StateStore.SetStates(blockHash, totalDelta, blockHash => this[blockHash]);
+                StateStore.SetStates(blockHash, totalDelta, blockHash =>
```

Comment:
How about changing the variable name `blockHash` one of the two? I think it could be confusing.

Reply:
I agree with your thought. It should be changed to another name and I will do it with the shortcut name like 'bh'

But I felt the origin of this issue was in passing getter function while I fixed this bug. The right getter was passed to it in test code but the real code (i.e. in `BlockChain<T>.SetStates(…)`) didn't. So, I felt it is a pattern that can cause mistakes easily. So I want to pass `Block<T>` as the first parameter and to remove the third parameter (i.e. getter function) because it has existed only to get previous hash. How do you have an opinion about this? Can it cause a problem in terms of design? 🤔

Reply:
I think that would be better.